### PR TITLE
[RFC] Coverity fixes

### PR DESCRIPTION
**Pull request by sander2**
Some coverity fixes. Since these bugs were known, this is a case of technical debt.

### CODE REVIEW

File: src/nvim/getchar.c [hunk 1]
```diff
@@ -1913,59 +1913,31 @@ static int vgetorpeek(int advance)
...
+            // matches at least what the matching mapping matched
+            keylen = 0;
+            // If there was no mapping, use the character from the typeahead
+            // buffer right here. Otherwise, use the mapping (loop around).
+            if (mp == NULL) {
+              // get a character: 2. from the typeahead buffer
+              c = typebuf.tb_buf[typebuf.tb_off] & 255;
+              if (advance) {                  // remove chars from tb_buf
+                cmd_silent = (typebuf.tb_silent > 0);
+                if (typebuf.tb_maplen > 0)
```
Comment (Hjdskes) [inline src/nvim/getchar.c]: Style: needs braces

File: src/nvim/getchar.c [hunk 2]
```diff
@@ -1913,59 +1913,31 @@ static int vgetorpeek(int advance)
...
+                  KeyTyped = FALSE;
+                else {
+                  KeyTyped = TRUE;
+                  // write char to script file(s) 
+                  gotchars(typebuf.tb_buf + typebuf.tb_off, 1);
                 }
-                break;                      /* got character, break for loop */
+                KeyNoremap = typebuf.tb_noremap[
+                  typebuf.tb_off];
```
Comment (Hjdskes) [inline src/nvim/getchar.c]: Style?

File: src/nvim/option.c [hunk 3]
```diff
@@ -1490,7 +1490,7 @@ do_set (
               new_value_alloced = true;
               if (newval == NULL) {
                 newval = empty_option;
-              } else if (!(options[opt_idx].flags | P_NO_DEF_EXP)) {
+              } else if (!(options[opt_idx].flags & P_NO_DEF_EXP)) {
```
Comment (Hjdskes) [inline src/nvim/option.c:1493]: Are we sure this is the correct behaviour?

**Review by Hjdskes — CHANGES_REQUESTED** [review]
Nice work!! Just two minor style fixes and a question :)

File: src/nvim/option.c (hunk 3 above)
Comment (sander2) [reply src/nvim/option.c:1493]: Pretty sure, since (1) this is how flags are generally used, (2) this is how flags are used in the rest of the file.

**Review by Hjdskes — APPROVED** [review]

**Comment by sander2** [comment]
Squashed the last commit.

**Comment by Hjdskes** [comment]
LGTM, merging.

**Pull request merged**